### ISSUE TITLE
Convenience class for adapters getting parameter views

### ISSUE THREAD

**Issue by swillner** [openclimatedata/openscm#175]

**Comment by znicholls** [comment]
+1 from me.

In #176 I added two attributes to `Adapter` to do this.

```python
_parameter_views: Dict[HierarchicalName, ParameterInfo]
"""Parameter views available with this model"""

_parameter_versions: Dict[HierarchicalName, int]
"""Parameter versions (last time parameter was passed to the model)"""
```

These are handy for two reasons. 

1. If you've set your views up right, you can get values of whatever by simply doing `self._parameter_views[hierarchical_name].value` 
    - or `.values` in the case of timeseries but I would advocate that we make it `.value` for all views to make life easy
1. You can check whether things have changed since you last set up your model by simply examining whether `self._parameter_views[hierarchical_name].version > self.__parameter_versions[hierarchical_name]`. For an incorporated adapter like DICE this obviously doesn't matter, as the model's parameters are always based on the parameter views. However for something else, e.g. MAGICC, yo want to do the slow step of updating the model (in MAGICC's case input files) as few times as possible and doing the check gives you a cheap way to decide whether this expensive step is necessary or not. 
    - incidentally, if you have a mechanism like this, you can get rid of `_initialize_model_input` and `_initialize_run_parameters` entirely. You just update if required and otherwise don't. Checking the required updates is still fast because looping over the parameterset and checking all the views is so computationally cheap.

**Comment by swillner** [comment]
I like the `_parameter_views`, but would implement them as attributes rather than dicts (though, that's just a matter of taste). Then the parameter views could easily be initialized using a dict with the defaults exactly like in the `DICE` adapter.
We don't need an extra `_parameter_versions` as every view saves the current version it buffers, so we can easily add a property
```
@property
def outdated(self):
    return self._version < self._parameter.version
```
to the view classes (maybe re-introduce a `ParamterView` base class for that). Could do so when I have a free minute.

**Comment by znicholls** [comment]
> would implement them as attributes rather than dicts

Attributes of the adapter or of `_parameter_views` i.e. would you do `self._ecs.value` or `self._parameter_views.ecs.value`?

> We don't need an extra `_parameter_versions` as every view saves the current version it buffers, so we can easily add a property

Nice

**Comment by swillner** [comment]
> Attributes of the adapter or of _parameter_views i.e. would you do self._ecs.value or self._parameter_views.ecs.value? I can do that in #186

`self._parameter_views.ecs.value` ;)

**Comment by znicholls** [comment]
ok so named tuple works too then yes?

**Comment by swillner** [comment]
yes. you could simply just generalize the approach in the DICE adapter.

**Comment by znicholls** [comment]
> you could simply just generalize the approach in the DICE adapter

sadly you can't do this as a namedtuple can only have strings as keys but we want to have tuples as keys too so we can distinguish between model parameters and openscm parameters.
